core/storage/email/gae_models_test.py

```diff
@@ -128,6 +149,168 @@ def test_get_by_hash_behavior_with_sent_datetime_lower_bound(self):
                     sent_datetime_lower_bound='Not a datetime object')
 
 
+class ReplyToIdModelUnitTests(test_utils.GenericTestBase):
+    """Test the GeneralFeedbackEmailReplyToIdModel class."""
+
+    def setUp(self):
+        super(ReplyToIdModelUnitTests, self).setUp()
+
+        def _generate_unique_id_for_tests(unused_cls):
+            return 'Non_unique_id'
+
+        self.generate_constant_id_ctx = self.swap(
```

Comment:
Is this necessary? It's generally better to test code from their APIs and not expose, or rely on, and specific internals. That way, the implementation of the interface can be replaced without the tests needing to be updated.

Is there a way for these tests to be written such that this swap isn't needed?

Reply:
It's only needed (and used) to test that when ids are non-unique, it correctly raise an exception. So the swap is only performed in one of the tests. I can't test the case otherwise I think

Reply:
It's not much better, but could we instead swap out ``get_random_int``? Or, we could introduce a custom ``utils`` function for ID generation and swap that to make it clear for other tests what function to replace. That doesn't let you override the generated value directly, but it does allow you to "influence" the generation such that the same ID is created twice.

Alternatively, you could seed the randomness before each ID is generated (which should reset the RNG, and in turn result in the same ID being generated twice).

Either approach I think would prevent you from needing to do this particular override.